webapp-django/crashstats/base/jinja2/crashstats_base.html

```diff
@@ -171,18 +166,13 @@ <h1>Product Navigation</h1>
                     <a href="{{ url('manage:home') }}">Admin</a>
                     {% endif %}
                     |
-                    {% if request.session.get('signin_method') == 'google' %}
+                    <form action="/oidc/logout/" style="display: inline" method="POST">
                         {% csrf_token %}
-                        <a href="{{ url('auth:oauth2_signout') }}" class="google-signout">
-                            <span>Sign out</span>
-                        </a>
-                    {% endif %}
+                        <input class="text-submit-input" type="submit" value="Sign out"/>
+                    </form>
                 {% else %}
                     {% csrf_token %}
-                    <div
-                        id="signin2"
-                        class="google-signin"
-                        data-signin-url="{{ url('auth:oauth2_signin') }}"></div>
+                    <a href="{{ url('oidc_authentication_init') }}">Sign in</a>
```

Comment:
Logging in with this link redirects me back to the homepage. mozilla-django-oidc looks for a [query parameter (`next` by default) that specifies what URL to direct the user to after logging in](https://mozilla-django-oidc.readthedocs.io/en/stable/settings.html#OIDC_REDIRECT_FIELD_NAME).

For constructing a properly escaped querystring, we have a helper function called [`make_query_string`](https://github.com/mozilla-services/socorro/blob/master/webapp-django/crashstats/base/templatetags/jinja_helpers.py#L101):

```jinja
{{ url('foobar') }}?{{ make_query_string(param='value') }}
```

For getting the current URL as a value, the `request` variable is available in templates, and the [`build_absolute_uri`](https://docs.djangoproject.com/en/2.0/ref/request-response/#django.http.HttpRequest.build_absolute_uri) function is available on Django requests to get the current URL, including the querystring.

Reply:
My expectation (that I haven't confirmed) would be that the `action` URL in the form for logging out also takes a `next` parameter.